src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -270,6 +270,7 @@ public void initializeMasterKey(ActionListener<Boolean> listener) {
                     }));
 
                 } else {
+                    context.restore();
```

Comment:
Can have this after if/else block.

Reply:
this doesn't work as we want to restore context before listener.onResponse happens here. Tested with this and we got some exceptions